videostore/utils.c

```diff
@@ -1,24 +1,51 @@
 #include "utils.h"
 
-int get_video_duration(int64_t *duration,   // OUT
-                       const char *filename // IN
+int get_video_info(VideoInfo *info,     // OUT
+                   const char *filename // IN
 ) {
+    info->duration = 0;
+    info->width = 0;
+    info->height = 0;
+    info->codec[0] = '\0';
+
     AVFormatContext *fmt_ctx = NULL;
     int ret;
+
     if ((ret = avformat_open_input(&fmt_ctx, filename, NULL, NULL)) < 0) {
         return ret;
     }
     if ((ret = avformat_find_stream_info(fmt_ctx, NULL)) < 0) {
         avformat_close_input(&fmt_ctx);
-    return ret;
+        return ret;
+    }
+
+    info->duration = fmt_ctx->duration;
+    if (info->duration == AV_NOPTS_VALUE) {
+        avformat_close_input(&fmt_ctx);
+        return VIDEO_STORE_VIDEO_INFO_RESP_ERROR;
+    }
+
+    for (unsigned i = 0; i < fmt_ctx->nb_streams; i++) {
+        AVStream *st = fmt_ctx->streams[i];
+        if (st->codecpar->codec_type == AVMEDIA_TYPE_VIDEO) {
+            info->width  = st->codecpar->width;
+            info->height = st->codecpar->height;
+            const char *codecName = avcodec_get_name(st->codecpar->codec_id);
+            if (codecName) {
```

Comment:
if this is nil, we should error right? Can we please just log & error here if this condition happens

Reply:
I think we are ok to error out when checking if codec is filled after the loop.